### ISSUE TITLE
Validate missing props in subclassed React components

### ISSUE THREAD

**Issue by Bloomca** [yannickcr/eslint-plugin-react#1220]
Hi!

Thanks for the awesome work!

We extend components with our own parent class to add few functions in it. However, we'd like to have missing props validation inside it. I've searched, and found only mentioning about it in [docs](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md#about-component-detection). Sorry if I missed it, but is it possible somehow to specify which subclasses are fine?

And just a basic example:
```javascript
class Hello extends CustomComponent {
  render() {
     return (
         <div>
             {this.props.message} // this should show an error from eslint
         </div>
     );
  }
}
```

**Comment by ljharb** [comment]
React components should never inherit from a "base class" - only from React.Component/React.PureComponent.

**Comment by Bloomca** [comment]
Why not? I am not talking about best practices or something, I am just asking whether it is possible or not to add missing props validation to custom extended classes.

**Comment by ljharb** [comment]
Why not, because inheritance is an inferior form of code reuse than composition, and even the React team strongly discourages it.

The workaround for you here is to use a jsdoc-style "extends" comment to annotate your `class` that it's a React component. I'm not sure of the exact syntax off the top of my head.

**Comment by jseminck** [comment]
To expand upon ljharbs comment, and for anyone else reading this in the future. This syntax is supported:

```
/**
 * @extends React.Component
 */
export default class Hello extends MySuperClass {
   ...
}
```

And

```
/**
 * @augments React.Component
 */
export default class Hello extends MySuperClass {
   ...
}
```

**Comment by Bloomca** [comment]
@jseminck thanks, it works!

Maybe we should add it in the readme of this rule? I understand that it is not a preferred way to extend functionality in React, but maybe it will be helpful for other people as well.
